src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -45,6 +46,9 @@ import { CancellationToken } from 'vs/base/common/cancellation';
 import 'sql/workbench/contrib/welcome/page/browser/az_data_welcome_page'; // {{SQL CARBON EDIT}}
 import { IHostService } from 'vs/workbench/services/host/browser/host';
 import { IProductService } from 'vs/platform/product/common/productService';
+// eslint-disable-next-line code-layering
```

Comment:
This rule should never be disabled. And in general don't just disable rules without checking that it's ok - a lot of these rules are set up so that things work correctly in our particular environment. 

For this specific change how important is this? Because this gets trickier once we consider that ADS isn't always going to be running in an environment with access to the file system (for example web mode).

Reply:
Hey Charles, yeah I found that out the hard way. I spoke with Anthony about this. He said there is a specific service to use here instead:

https://github.com/microsoft/azuredatastudio/blob/master/src/vs/platform/files/common/files.ts#L87

Reply:
I believe this has been fixed